qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -139,6 +132,11 @@ def __init__(
                 These are: the evaluation count, the optimizer parameters for the
                 ansatz, the evaluated mean and the evaluated standard deviation.`
             quantum_instance: Quantum Instance or Backend
+            sort_parameters_by_name: If True, the initial point is bound to the ansatz parameters
```

Comment:
Are we going to drop the old way of sorting. I think this sorting still causes people issues when trying to use the anstaz themselves. Now by going with False I guess there can already some sort of breakage in compatibility - but I guess the option is there to set to True to avoid that. But maybe we should already deprecate doing that (i.,e detect when True) so this flag can go away (maybe it would be easier to make it go away it it were optional so we detect any usage of it). This way the params and the natural sort order things should always work even if put the final optimal point directly to the ansatz right.

Reply:
I'll add a deprecation 👍🏻